### PR TITLE
Consumer tests

### PR DESCRIPTION
**Pull request by tisnik**
# Description

More thorough unit tests for Consumer interface.

## Type of change

- Unit tests (no changes in the code)

## Testing steps
Please run `make test` as usual.

### CODE REVIEW

File: consumer/consumer.go [hunk 1]
```diff
@@ -102,17 +100,24 @@ func (consumer Impl) Start() error {
...
-		}
-		err = consumer.Storage.WriteReportForCluster(orgID, clusterName, report)
-		if err != nil {
-			log.Println("Error writing report to database:", err)
-		}
+	}
+}
+
+// ProcessMessage processes an incoming message
+func (consumer Impl) ProcessMessage(msg *sarama.ConsumerMessage) {
```
Comment (JoseLSegura) [inline consumer/consumer.go]: As totally newbie point of view:
Why don't returning an `error` too? When an `err` is returned from the parsing of the message, you can "inform" about it to the caller

File: consumer/consumer_test.go [hunk 2]
```diff
@@ -47,3 +49,164 @@ func TestConsumerConstructorNoKafka(t *testing.T) {
...
+	}
+}
+
+func TestParseMessageWithImproperJSON(t *testing.T) {
+	const message = `"this_is_not_json_dude"`
+	_, _, _, err := consumer.ParseMessage([]byte(message))
+	if err == nil {
+		t.Fatal("Error is expected to be returned for empty message")
+	}
+}
```
Comment (JoseLSegura) [inline consumer/consumer_test.go]: Can the `err` be checked in order to see if the error is the expected one?

File: consumer/consumer_test.go [hunk 3]
```diff
@@ -47,3 +49,164 @@ func TestConsumerConstructorNoKafka(t *testing.T) {
...
+
+func TestParseMessageWithWrongContent(t *testing.T) {
+	const message = `{"this":"is", "not":"expected content"}`
+	_, _, _, err := consumer.ParseMessage([]byte(message))
+	if err == nil {
+		t.Fatal("Error is expected to be returned for empty message")
```
Comment (JoseLSegura) [inline consumer/consumer_test.go]: This error string can be confusing, because the `message` is incorrect, but not empty

Comment (tisnik) [reply consumer/consumer_test.go]: oh good catch!

File: consumer/consumer_test.go (hunk 2 above)
Comment (tisnik) [reply consumer/consumer_test.go]: you are right, I'll fix it.

File: consumer/consumer.go (hunk 1 above)
Comment (tisnik) [reply consumer/consumer.go]: yes, we can do it. thanks for review!

Comment (JoseLSegura) [reply consumer/consumer.go]: :-) Glad to help

File: consumer/consumer.go [hunk 4]
```diff
@@ -102,20 +100,29 @@ func (consumer Impl) Start() error {
 	consumed := 0
 	for {
 		msg := <-consumer.PartitionConsumer.Messages()
-		log.Printf("Consumed message offset %d\n", msg.Offset)
+		consumer.ProcessMessage(msg)
```
Comment (JoseLSegura) [inline consumer/consumer.go]: Probably you want to check if `ProcessMessage` returned an error

Comment (tisnik) [reply consumer/consumer.go]: fixed, nice catch!

**Pull request merged**